File: frontend/src/api/score.ts
import client from './client'

export interface ScoreFactor {
  key: string
  value: number
  impact: 'positive' | 'negative' | 'neutral'
  description_ru: string
}

export interface ScoreResponse {
  score: number
  delta: number
  grade: string
  grade_label: string
  fetched_at: string
  factors: ScoreFactor[]
}

export interface ScoreHistoryItem {
  date: string
  score: number
  delta: number
}

export interface ScoreHistoryResponse {
  history: ScoreHistoryItem[]
}

export async function getCurrentScore(): Promise<ScoreResponse> {
  const response = await client.get<ScoreResponse>('/score/current')
  return response.data
}

export async function getScoreHistory(): Promise<ScoreHistoryResponse> {
  const response = await client.get<ScoreHistoryResponse>('/score/history')
  return response.data
}

export async function getScoreFactors(): Promise<ScoreFactor[]> {
  const response = await client.get<ScoreFactor[]>('/score/factors')
  return response.data
}

File: backend/app/schemas/score.py
from pydantic import BaseModel
from datetime import datetime
from typing import List, Optional


class FactorResponse(BaseModel):
    key: str
    value: float
    impact: str  # positive, negative, neutral
    description_ru: str

    class Config:
        from_attributes = True


class ScoreResponse(BaseModel):
    score: int
    delta: int
    grade: str
    grade_label: str
    fetched_at: datetime
    factors: List[FactorResponse] = []

    class Config:
        from_attributes = True


class ScoreHistory(BaseModel):
    date: datetime
    score: int
    delta: int

    class Config:
        from_attributes = True


class ScoreHistoryResponse(BaseModel):
    history: List[ScoreHistory]
